app/src/main/java/com/T05/krowdtrialz/ui/ExperimentDetailsNonOwnerActivity.java

```diff
@@ -44,15 +60,121 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_experiment_details_non_owner);
+        db = Database.getInstance();
 
         Intent intent = getIntent();
         // TODO: Use this to get experiment object
         String experimentID = intent.getStringExtra(SubscribedFragment.EXTRA_EXPERIMENT_ID);
 
+        db.getExperimentByID(experimentID, new Database.GetExperimentCallback() {
+            @Override
+            public void onSuccess(Experiment exp) {
+                experiment = exp;
+            }
+
+            @Override
+            public void onFailure() {
+                Log.e(TAG, "Error Searching Database for experiment");
+            }
+        });
+
+        populateMainInfo();
```

Comment:
I believe these 4 lines should be in onSuccess. This is because the only time we have a guarantee that the query getExperimentByID has completed is once we are in onSuccess. Simply putting these lines after the listener is created does not guarantee the query is complete, and will likely result in null reference errors.